Assets/LeapMotionModules/FramerateGraph/Scripts/FramerateGraph.cs

```diff
@@ -0,0 +1,227 @@
+
+    private Texture2D _gradientTexture;
+
+    private bool _hasCompletedFrame = false;
+
+    private long _prevEofTick = 0;
+    private long _preRenderTick = 0;
+
+    private long _totalRenderTick = 0;
+    private long _totalFrameTick = 0;
+
+    private long _updateStartTick = -1;
+    private long _updateEndTick = -1;
+    private long _totalUpdateTick = -1;
+
+    private SmoothedFloat _smoothedFrameTime = new SmoothedFloat();
+    private SmoothedFloat _smoothedRenderTime = new SmoothedFloat();
+    private SmoothedFloat _smoothedUpdateTime = new SmoothedFloat();
+
+    private long _lastSpikeTick = 0;
+    private SmoothedFloat _smoothedSpikeRate = new SmoothedFloat();
+
+    void Awake() {
+      _smoothedFrameTime.delay = _smoothingDelay;
+      _smoothedRenderTime.delay = _smoothingDelay;
+      _smoothedUpdateTime.delay = _smoothingDelay;
+      _smoothedSpikeRate.delay = _spikeSmoothingDelay;
+
+      _graphTexture = new Texture2D(_frameResolution, 1, TextureFormat.ARGB32, false, true);
+      _graphTexture.filterMode = FilterMode.Point;
+      _graphTexture.wrapMode = TextureWrapMode.Repeat;
+
+      Color32[] black = new Color32[_frameResolution];
+      for (int i = 0; i < black.Length; i++) {
+        black[i] = Color.black;
+      }
+      _graphTexture.SetPixels32(black);
+      _graphTexture.Apply();
+
+      _graphRenderer.material = new Material(_graphShader);
+      _graphRenderer.material.mainTexture = _graphTexture;
+
+      _gradientTexture = new Texture2D(256, 1, TextureFormat.ARGB32, false, true);
+      _gradientTexture.filterMode = FilterMode.Bilinear;
+      _gradientTexture.wrapMode = TextureWrapMode.Clamp;
+
+      Color32[] colors = new Color32[256];
+      for (int i = 0; i < colors.Length; i++) {
+        colors[i] = _colorGradient.Evaluate(i / 256.0f);
+      }
+      _gradientTexture.SetPixels32(colors);
+      _gradientTexture.Apply();
+
+      _graphRenderer.material.SetTexture("_Ramp", _gradientTexture);
+
+      _cachedStrings = new string[500];
+      for (int i = 0; i < 500; i++) {
+        _cachedStrings[i] = (i / 10.0f).ToString();
+      }
+    }
+
+    void OnEnable() {
+      _stopwatch.Reset();
+      _stopwatch.Start();
+      StartCoroutine(endOfFrameCoroutine());
+
+      Camera.onPreRender += onPreRender;
+      Camera.onPostRender += onPostRender;
+
+      _hasCompletedFrame = false;
+
+      _smoothedFrameTime.reset = true;
+      _smoothedRenderTime.reset = true;
+      _smoothedUpdateTime.reset = true;
+    }
+
+    void OnDisable() {
+      _stopwatch.Stop();
+      StopAllCoroutines();
+
+      Camera.onPreCull -= onPreRender;
+      Camera.onPostRender -= onPostRender;
+    }
+
+    void FixedUpdate() {
+      if (_updateStartTick == -1) {
+        _updateStartTick = _stopwatch.ElapsedTicks;
+      }
+    }
+
+    void Update() {
+      if (_hasCompletedFrame) {
+        float totalRenderMilis = _totalRenderTick / (float)System.Diagnostics.Stopwatch.Frequency * 1000.0f;
```

Comment:
Technically these should all be "Millis", too, but I don't know if we care about the spelling of local variables that much.